c++/src/vector.cc

```diff
@@ -11,6 +11,7 @@ class Vector {
   unsigned int length_ = 0;
   unsigned int capacity_ = 8;
   T* dataStore_ = nullptr;
+  const unsigned int growFactor = 2;
```

Comment:
A cool thing you can do, though I'm not sure how useful it would be, would be to remove this variable and make it a template parameter:

``` cpp
template <class T, std::size_t F = 2>
class Vector { ... }
```

Then you could just refer to `F` wherever you would've referred to `GROWTH_FACTOR`. Then it could be configured when it's created:

``` cpp
Vector<int, 4> fourFactorVector{1, 2, 3};
```

This is for example how `std::array` lets you specify the length of a stack-allocated array, because the type parameter is statically defined as opposed to dynamically defined.

That said, I still don't think I would do this because it's not all that useful to configure this externally; it's an implementation detail. I would keep it how you have it.